includes/controller/angeltypes_controller.php

```diff
@@ -92,7 +92,10 @@ function angeltype_edit_controller()
         // Edit existing angeltype
         $angeltype = AngelType::findOrFail($request->input('angeltype_id'));
 
-        if (!auth()->user()->isAngelTypeSupporter($angeltype) && !auth()->can('admin_user_angeltypes')) {
+        if (
+            !auth()->user()
+            || (!auth()->user()->isAngelTypeSupporter($angeltype) && !auth()->can('admin_user_angeltypes'))
```

Comment:
The future is now! :rocket: 

This should work:

```suggestion
            !auth()->user()?->isAngelTypeSupporter($angeltype) && !auth()->can('admin_user_angeltypes')
```